src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -727,19 +727,18 @@ subroutine ocn_diagnostic_solve(dt, statePool, forcingPool, meshPool, diagnostic
       !
 
       nCells = nCellsArray(3)
-      !$omp do schedule(runtime) private(invAreaCell1, k, shearSquared, i, iEdge, factor, delU2, shearMean)
+      !$omp do schedule(runtime) private(invAreaCell1, k, shearSquared, i, iEdge, delU2, shearMean)
       do iCell=1,nCells
          RiTopOfCell(:,iCell) = 100.0_RKIND
-         invAreaCell1 = 1.0_RKIND / areaCell(iCell)
          do k=2,maxLevelCell(iCell)
            shearSquared = 0.0_RKIND
            do i = 1, nEdgesOnCell(iCell)
              iEdge = edgesOnCell(i, iCell)
-             factor = 0.5_RKIND * dcEdge(iEdge) * dvEdge(iEdge) * invAreaCell1
              delU2 = (normalVelocity(k-1,iEdge) - normalVelocity(k,iEdge))**2
-             shearSquared = shearSquared + factor * delU2
+             shearSquared = shearSquared + edgeAreaFractionOfCell(i,iCell) * delU2
```

Comment:
In this PR I am only changing from `factor` to `edgeAreaFractionOfCell` in what was originally changed in this PR, since these are corrections. I tested before versus after this substitution, and it is not bfb but machine precision (1e-13 or less). I will change them in the rest of the code in a different PR.

Reply:
Probably not surprising that the code wouldn't be bit-for-bit given potential changes in order of operations. That sounds reasonable to me but we would also need to make sure the results aren't climate changing, unless the bug fixes themselves are climate changing (entirely possible).

Reply:
@mark-petersen this is consistent with what I saw in testing, the sum of the triangles does not quite equal areaCell, it is off at about the level you show here (1e-13).  Xylar could be right here too, that it isn't the triangle area itself but order of operations.